SCClassLibrary/Common/Core/Error.sc

```diff
@@ -191,6 +191,56 @@ DoesNotUnderstandError : MethodError {
 	}
 }
 
+DoesNotUnderstandWithKeysError : MethodError {
+	var <>selector, <>argumentsArray, <>keywordArgumentPairs, <suggestedCorrection, suggestion = "";
+	*new { arg receiver, selector, argumentsArray, keywordArgumentPairs;
+		^super.new(nil, receiver)
+		.selector_(selector)
+		.argumentsArray_(argumentsArray)
+		.keywordArgumentPairs_(keywordArgumentPairs)
+		.init
+	}
+
+	init {
+		var methods, methodNames, editDistances, minIndex, lowerCaseSelector;
+
+		if(receiver.notNil) {
+			methods = receiver.class.superclasses.add(receiver.class).collect(_.methods).reduce('++');
+			methodNames = methods.collect { |x| x.name.asString.toLower };
+			// we compare lower-case versions to prioritize capitalization mistakes
+			lowerCaseSelector = selector.asString.toLower;
+			editDistances = methodNames.collect(_.editDistance(lowerCaseSelector));
+			minIndex = editDistances.minIndex;
+			// Edit distance of 3 chosen arbitrarily; also, filter out completely dissimilar matches
+			// to avoid unhelpful suggestions.
+			if(editDistances[minIndex] <= 3 and: { methodNames[minIndex].similarity(lowerCaseSelector) > 0 }) {
+				suggestedCorrection = methods[minIndex];
+				suggestion = "\nPerhaps you misspelled '%', or meant to call '%' on another receiver?"
+				.format(suggestedCorrection.name, selector);
+			}
+		}
+	}
+	errorString {
+		^"ERROR: Message '" ++ selector ++ "' not understood." ++ suggestion
+	}
+	reportError {
+		this.errorString.postln;
+		"RECEIVER:\n".post;
+		receiver.dump;
+		"ARGS WITHOUT KEYS:\n".post;
+		argumentsArray.dumpAll;
+		"ARGS WITH KEYS:\n".post;
+		keywordArgumentPairs.dumpAll;
+		this.errorPathString.post;
+		if(protectedBacktrace.notNil, { this.postProtectedBacktrace });
+		this.dumpBackTrace;
+		// this.adviceLink.postln;
```

Comment:
is this going to be used? otherwise remove this line

Reply:
Its used in Object.doesNotUnderstandWithKeys

Reply:
mh, maybe I misunderstood, but this `adviceLink` ist currently commented out, I was wondering if this line can be removed (following the idea of: no working code in comments)

Reply:
Ah, no I thought you meant the whole class. 

I just copied what DoesNotUnderstandError does, but looking at it a few weeks later, I'm going to remove this class.